Get register of interest via API

ajparsons opened the pull request:
Parliament's turned off their old site, so our scraper is broken!

This scopes out the old parser, adds a new parser that converts the Parliament API into the XML TheyWorkForYou expects.

As seen in `dailyupdate`, this can be run via:

```bash
poetry run python -m pyscraper.regmem download-all-registers --chamber commons
```

This updates some requirements in the poetry, and also *uses* this poetry config for the first time in production.

Rather than exactly replicating the previous structure, there's now a bit more use of lists to reflect how parliament sees items and subitems.

I've tested this in both the import to database:

![image](https://github.com/user-attachments/assets/a2632381-b89c-4770-ae4f-a90501bebf77)


And the comparison over time:

![image](https://github.com/user-attachments/assets/ce5c2bdf-35e1-4258-af62-ab510f87e5ed)


There's a boundary problem in that the sudden change will break the retrospect. If we delete and recreate the last few items though, this should be less of a problem. 

Also have the option if reviewing this will take a while to just upload the generated files manually (given they seem fine).

ajparsons commented:
twfy host now has poetry - so this should work even in advance of the twfy poetry pr.

dracos commented on pyscraper/regmem/__main__.py:
```diff
@@ -0,0 +1,58 @@
+    chamber: str, date: datetime.date, force_refresh: bool = False, quiet: bool = False
+):
+    if chamber == "commons":
+        commons_process.download_register_from_date(
```
why does for_date become from_date :)

ajparsons replied:
Yep, made the same.

dracos commented on pyscraper/regmem/commons/api_models.py:
```diff
@@ -0,0 +1,225 @@
+from enum import Enum
+from typing import Any, List, Optional
+
+from jinja2 import Template
```
jinja2 is not in the requirements

dracos commented on pyscraper/regmem/commons/process.py:
```diff
@@ -0,0 +1,383 @@
+
+    # given we pull in parent interests when not in the category, we need to check we haven't duplicated
+
+    existing_ids = []
```
Should these be sets, if I'm understanding it right? Could get same ID more than once

ajparsons replied:
Yeah, it's not big enough for a real performance penalty but makes it clearer they should be unique.

dracos reviewed (changes requested):
One issue with missing jinja2. To be honest, I didn't follow all the code (perhaps some doctests of e.g. move_subitems_under_parent to show an example would be nice, not sure I understood the "parents" stuff), but if you're happy with the output, fine with me :)

ajparsons commented:
Added above - added some more commentary on the subitem movements because that wasn't clear. 

Added one extra fix that was needed for a march register to manually fetch a not present parent.

dracos reviewed (approved):
Thanks!
